spec/models/concerns/digital_object_concerns/attribute_assignment/doi_spec.rb

```diff
@@ -71,4 +72,31 @@
       end
     end
   end
+  describe "#ensure_doi_if_requested" do
+    let(:mint_doi) { true }
+    before { digital_object_with_doi.mint_doi = mint_doi }
+    context "when a doi has already been set" do
+      it "does not call ensure_doi in save hook" do
+        expect(Hyacinth::Config.external_identifier_adapter).not_to receive(:mint)
+        digital_object_with_doi.save
+        expect(digital_object.mint_doi).to be false
```

Comment:
Is the mismatch digital_object_with_doi (line 81) and digital_object (line 82) purposeful?

Reply:
No, I think that's an error! Will fix.